test: add integ tests for advanced count fields

SoManyHs opened the pull request:
Fixes https://github.com/aws/copilot-cli/issues/2218

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

SoManyHs committed:
test: add advanced count overrides to integ test

SoManyHs committed:
fix(manifest): correctly set AdvancedConfig with env overrides

SoManyHs committed:
chore: update go.sum

efekarakus commented on internal/pkg/manifest/lb_web_svc.go:
```diff
@@ -206,6 +206,11 @@ func (s LoadBalancedWebService) ApplyEnv(envName string) (*LoadBalancedWebServic
+	envCount := overrideConfig.TaskConfig.Count
+
+	if !envCount.IsEmpty() {
+		s.TaskConfig.Count = envCount // will this mess up the other envs?
```
😂 this made me laugh lol 

Can we add this as a unit test where we apply an `envName` to the override and see if the original count remains intact although the returned object changed?

I think this will be safe because the receiver is a concrete type `(s LoadBalancedWebService)` and not `(s *LoadBalancedWebService)` so the original object shouldn't be touched. But we should definitely validate this

efekarakus replied:
Do we need to make the same changes for Backend Service as well?

SoManyHs replied:
gah, yes. keep forgetting that -- would be nice if we could dedupe some of that logic!

SoManyHs replied:
Anyway, I was just pushing up https://github.com/aws/copilot-cli/pull/2223/commits/577c9b7030ad1c5abe44759c7844f0bee23f90df which should address your first comment!

SoManyHs committed:
chore: rename test files sensibly

efekarakus reviewed (commented):
The code looks good!

SoManyHs committed:
test: add test case with count not overridden in environment

Adding a new environment, `foo`, in the manifest that does not specify
an advancedCount override ensures that the count and autoscaling fields
are still using the default ones specified in the top-level `count`
fields.

SoManyHs committed:
chore: remove silly comment

Should be addressed by 577c9b70

SoManyHs committed:
chore: update BackendService.ApplyEnv

Uses same logic as lb-web-svc

SoManyHs committed:
chore: remove count value in backend service test

Value and AdvancedCount are mutually exclusive under the `Count` field.

SoManyHs committed:
Merge branch 'mainline' into spot-integ

Lou1415926 reviewed (approved):
The code 👀 GTM! Nothing much to add. Thank you for the unit tests on count!
